#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Define __libc_cleanup_pop and __libc_cleanup_pop
# DP: Related bugs: 
# DP: Author: Alfred M. Szmidt
# DP: Upstream status: In CVS
# DP: Status Details: 
# DP: Date: 2003-12-31

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

2003-08-07  Alfred M. Szmidt  <ams@kemisten.nu>
 
        * sysdeps/mach/hurd/bits/libc-lock.h
        (__libc_cleanup_push, __libc_cleanup_pop): New macros.
 
2003-09-28  Alfred M. Szmidt  <ams@kemisten.nu>

      * sysdeps/mach/hurd/bits/libc-lock.h (__libc_cleanup_pop):
      Redefine and use __libc_cleanup_region_end instead.

# append the patch here and adjust the -p? flag in the patch calls.
Index: sysdeps/mach/hurd/bits/libc-lock.h
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/bits/libc-lock.h,v
retrieving revision 1.2
retrieving revision 1.4
diff -u -p -r1.2 -r1.4
--- sysdeps/mach/hurd/bits/libc-lock.h	6 Dec 2002 11:33:53 -0000	1.2
+++ sysdeps/mach/hurd/bits/libc-lock.h	29 Sep 2003 21:55:41 -0000	1.4
@@ -1,5 +1,5 @@
 /* libc-internal interface for mutex locks.  Hurd version using Mach cthreads.
-   Copyright (C) 1996,97,98,2000,01, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1996,97,98,2000,01, 2002, 2003 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -159,6 +159,8 @@ typedef struct __libc_lock_recursive_opa
   if ((DOIT) && __save_FCT != 0)					    \
     (*__save_FCT)(__save_ARG);						    \
 
+#define __libc_cleanup_push(fct, arg) __libc_cleanup_region_start (1, fct, arg)
+#define __libc_cleanup_pop(execute) __libc_cleanup_region_end (execute)
 
 #if (_CTHREADS_ - 0)
 
